src/test/java/io/openliberty/tools/intellij/it/TestUtils.java

```diff
@@ -562,4 +571,41 @@ public static void copyDirectory(String sourceDirectoryLocation, String destinat
                 });
     }
 
+    /**
+     * Test to see if there has been a fatal error and JUnit should be stopped.
+     * This searches the output of this JUnit run for SocketTimeoutException which
+     * has been identified as a fatal error and occurs during the Mac tests.
+     * It is set up as a static reader so that we do not reread the whole file
+     * after each test.
+     */
+    final static String outputFile = System.getenv("JUNIT_OUTPUT_TXT");
+    final static BufferedReader reader;
+
+    static {
+        try {
+            reader = new BufferedReader(new InputStreamReader(new FileInputStream(outputFile)));
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void detectFatalError() {
+        try {
+            // Flush our output and then wait briefly for the process running 'tee' to flush output to the
+            // file that the buffered reader is reading.
+            System.out.flush();
+            Thread.sleep(10);
+            // Continue reading the existing reader
+            String line = null;
+            while ((line = reader.readLine()) != null) {
+                if (line.contains("SocketTimeoutException")) {
+                    System.exit(23);
```

Comment:
This is a magic number: https://en.wikipedia.org/wiki/Magic_number_(programming). Does this exit code actually mean anything? Are the scripts looking for this value? I didn't see any references in this PR.

Reply:
Would you like it better if I added a `public static int STE_FOUND = 23;` ?

Reply:
Actually, this value is not even passed by `gradle test` out to the environment. Gradle catches it and just exits with rc=1. So I think I'll just change this value to 1.